cli/src/pcluster/templates/cluster_stack.py

```diff
@@ -1422,7 +1422,8 @@ def _add_head_node(self):
                             f" cfn-signal --exit-code=0 --reason='Update complete'"
                             f" --region {self.stack.region} --url {cloudformation_url}"
                             f" '{self.wait_condition_handle.ref}' ||"
-                            f" cfn-signal --exit-code=1 --reason='Update failed'"
+                            f' cfn-signal --exit-code=1 --reason="$(cat /var/log/parallelcluster/bootstrap_error_msg '
+                            f"2>/dev/null || echo 'Update failed')\""
```

Comment:
we will need to remove this file if it exist at the end in case the next update failure doesn't write to the file

Reply:
Added.

Reply:
have you considered to handle the cases where the content of `bootstrap_error_msg` could be problematic? 
One I can think of is when the size is big enough to exceed the limit on the allowed parameter size.

Reply:
Good point! I think I can put a size limit of 4096 bytes.